navigation_and_routing/lib/src/auth/auth.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+/// A mock authentication service
+class BookstoreAuth extends ChangeNotifier {
+  bool _signedIn = false;
+
+  bool get signedIn {
+    return _signedIn;
+  }
+
+  Future<void> signOut() async {
+    await Future<void>.delayed(const Duration(milliseconds: 200));
+    // Sign out.
+    _signedIn = false;
+    notifyListeners();
+  }
+
+  Future<bool> signIn(String username, String password) async {
+    await Future<void>.delayed(const Duration(milliseconds: 200));
+
+    // Sign in. Allow any password.
+    _signedIn = true;
+    notifyListeners();
+    return _signedIn;
+  }
+
+  @override
+  bool operator ==(Object other) {
```

Comment:
Having custom `operator==` and `hashCode` on a non-immutable object is strange and can yield undesired/unpredictable behavior. 

See also: https://dart.dev/guides/language/effective-dart/design#avoid-defining-custom-equality-for-mutable-classes

Do you actually need this?